core/src/main/java/feign/querymap/FieldQueryMapEncoder.java

```diff
@@ -66,14 +67,11 @@ private ObjectParamMetadata(List<Field> objectFields) {
     }
 
     private static ObjectParamMetadata parseObjectType(Class<?> type) {
-      List<Field> fields = new ArrayList<Field>();
-      for (Field field : type.getDeclaredFields()) {
-        if (!field.isAccessible()) {
-          field.setAccessible(true);
-        }
-        fields.add(field);
-      }
-      return new ObjectParamMetadata(fields);
+      return new ObjectParamMetadata(
+          Arrays.stream(type.getDeclaredFields())
+              .filter(field -> !field.isSynthetic())
+              .peek(field -> field.setAccessible(true))
```

Comment:
Technically, this violates the non-interference contract, but I don't see any actual issues with it.